### ISSUE TITLE
Segmentation notebook works wrong?

### ISSUE THREAD

**Issue by Fautos** [torch-points3d/torch-points3d#633]
Hy.
This is the first time i´m writting an issue, so sorry if i make any mistake.

I´ve been looking at the notebook of segmentation using KPConv available here:
https://github.com/nicolas-chaulet/torch-points3d/blob/master/notebooks/PartSegmentationKPConv.ipynb
But when i run it it doesnt works properly.
I dont know where the problem is, but the net is not learning at all. Doesnt matter how many times i train the net, it allways output the same value.

Let me explain it better with images so you can understand what i´m saying.
![image](https://user-images.githubusercontent.com/55849905/123255553-46343c00-d4f0-11eb-91b1-da050f08fc7c.png)
*Note: The text is in Spanish, dont worry about it.
This image represent 4 outups of the net. As you can see all of them have more or less the same structure, being the second value the highest one. In the red square you can see the predicted output given by the net (in spanish "predicción") and the real output provided by the dataset (Real). It doesnt matter how many epoch i train the net it´s always the same structure. Moreover, the more i train the higher the values, but the structure is completly the same (being the second value the highest one, and the rest having more or lees the same difference)

In this image i train 5 epoch (as i said before it doesnt matter how many epoch you train, the output is always the same) an get the following logdir:
![image](https://user-images.githubusercontent.com/55849905/123256675-87791b80-d4f1-11eb-9f14-7703e3e166a3.png)
If you train the net more you will get more or less the same curves, but as you saw the output missmatch the graphics.

I think i havent change anything in the notebook, im using everything as it should be, so i cant understand why im getting this results (yes, i tried so many times) so the only thing i can do now is asking for your help.

I´m using Google collab, if you need to see anything in my code just ask for it, i´m not attaching the code because it´s long and more or less it´s the same as the one published here.

Thanks.

**Comment by nicolas-chaulet** [comment]
That's interesting, the image of the logs looks correct, the network performs at 75% IoU or so. Which is the expected outcome after a few epochs. So the problem is more with the text output than with the network itself I think.

**Comment by Fautos** [comment]
Thanks for answering Nicolas.
I agree about the logs, it looks like the performance should be very good, but i dont understand what do you mean by the text output. To get that output i used the validation set, which i load like this:
![image](https://user-images.githubusercontent.com/55849905/123285374-45f66980-d50d-11eb-991f-59b87f05054a.png)
I dont think the is any issue in this part because i can get the output easily using the following sentence:
`output=model.forward(val_samples[object])   #Where object is the index that represent the object we will study` 
"Output" is an array of tensors like the ones i show you in the previous message, to get the output of any point I code: 
`specific_output=output[point]  #Where point is the index to a specific point `
And to choose the prediction i take the highest value in the tensor.
Then I compare this output with the real one. I´m looking at the real one with the following sentence:
`val_samples[object].y[punto]  #Where object is the same as before, and point the index to a specif point in the could`

In my code I have created a function with the following inputs:
![image](https://user-images.githubusercontent.com/55849905/123292255-295d3000-d513-11eb-87e9-e7f17b3d44ac.png)

And the function is like this:
![image](https://user-images.githubusercontent.com/55849905/123292452-57db0b00-d513-11eb-86dc-863f9cee1f07.png)
![image](https://user-images.githubusercontent.com/55849905/123292550-6e816200-d513-11eb-80f6-7d2b9cd787ac.png)
![image](https://user-images.githubusercontent.com/55849905/123292583-793bf700-d513-11eb-8576-0cf4b9ef9f8d.png)

I ´m showing the complete function but the important lines are in the first and last pictures. As you can see in the first one I´m creating "resultado_real" which is the real output (val_samples[object].y) as a list so i can compare it. Then im getting the prediction in kinda complex way, but with that loop I can convert the tensor to list and check where the highest value is.
At the end, in the last picture i compare both the real result and the prediction, and that´s what you can see in the previous message.

The same idea I used in the classification notebook and there it works perfectly, so I think this should work, but maybe you can see where i´m failing or what I´m missunderstanding.

Again thanks for your help.
